src/PlaneWaveBasis.jl

```diff
@@ -316,6 +316,29 @@ Creates a new basis identical to `basis`, but with a custom set of kpoints
                    basis.symmetries_respect_rgrid, basis.comm_kpts)
 end
 
+"""
+    PlaneWaveBasis(basis, lattice, positions)
+    PlaneWaveBasis(basis, lattice)
+    PlaneWaveBasis(basis, positions)
+
+Creates a new basis, identical to `basis`, but different `lattice` or `positions`.
+"""
+function PlaneWaveBasis(basis::PlaneWaveBasis, lattice::AbstractMatrix,
+                        positions::Vector{<:AbstractVector}=basis.model.positions)
+    model = Model(basis.model, lattice, positions)
+    PlaneWaveBasis(model,
+                   basis.Ecut, basis.fft_size, basis.variational,
+                   basis.kcoords_global, basis.kweights_global,
+                   basis.kgrid, basis.kshift, basis.symmetries_respect_rgrid,
+                   basis.comm_kpts)
```

Comment:
Hm this can have unintended consequences, ie the model can change symmetries, lattice leading to different fft_size, etc...

Reply:
There are two possible intended behaviors here: either keep the basis as much the same as possible, or keep it as close to a new basis as possible. This sort of does neither. Not sure what to do here.

Reply:
I think it should be as close to the new basis as possible, i.e. it should be a new consistent basis with the new structure. So yes, I think this should change the FFT size if needed. I did not think about that.

Reply:
Then I'm not sure this can be done sensibly, because we have different ways of constructing a pwbasis. I think we should move towards encapsulating the kpoint grid (and possibly the G grid ?) better,and then have a unique pwbasis(::Model, ::Kgrid) constructor, which would then make what you want to do here more straightforward.

Reply:
By Kgrid I mean the specification (uniform/custom/symmetrized/shifted/..., with symmetries or not, not the actual instantiation) of the kgrid. Same for the G grid (fft_size, variational, )